project/SparkBuild.scala

```diff
@@ -420,6 +422,16 @@ object Unidoc {
   )
 }
 
+object Scalastyle {
```

Comment:
I've been trying to futz with this for a while, and I'm pretty stuck too.  I've tried assorted permutations and I can't get it to work -- *unless* I put a `build.sbt` file in *every* project sub-project, which just has:

```
scalastyleConfig := file("dev/scalastyle-config.xml")

```

What I find really strange is that nothing I did in `SparkBuild.scala` takes effect -- but I can modify the location interactively in the console.  If I go into a subproject, and then run

```
set scalastyleConfig := file("dev/scalastyle-config.xml")
```

then I can run `scalastyle` in that project and it will pick up the right settings.  However, doing this once in the project root does **not** effect all of the sub-projects.  You have to do it in each project separately.

I came up with the solution for using the `build.sbt` file in each project, just by applying that solution in the shell, and then executing `session save`, which creates that file.  Miraculously the settings really were saved -- run sbt again, and now each project has the right scalastyle settings.

That might be too hacky & convoluted to be worth this change ... but maybe it will give somebody else a clue to what is going on here ...